torchrl/data/datasets/d4rl.py

```diff
@@ -149,6 +158,21 @@ def __init__(
         )
         self.extend(dataset)
 
+    def _get_dataset_direct_download(self, name, env_kwargs):
+        """Directly download and use a D4RL dataset."""
+        if env_kwargs:
+            raise RuntimeError("Cannot pass env_kwargs when `direct_download=True`.")
+        url = D4RL_DATASETS.get(name, None)
+        if url is None:
+            raise KeyError(f"Env {name} not found.")
+        h5path = _download_dataset_from_url(url)
+        # h5path_parent = Path(h5path).parent
+        dataset = PersistentTensorDict.from_h5(h5path)
```

Comment:
It fails on the 'metadata' key, in the current D4RL implementation metadata is explicitly removed. So we either need to change peristent.py in tensordict, or write our processing here.

`
        if "metadata" in dataset.keys():
            metadata = dataset.get("metadata")
            dataset = dataset.exclude("metadata")
            self.metadata = metadata
            # find batch size
            dataset = make_tensordict(dataset.flatten_keys("/").to_dict())
            dataset = dataset.unflatten_keys("/")
`